x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -1079,83 +1080,155 @@ public void testMatchFilter() throws Exception {
+    }
+
+    private void checkFullTextFunctionNotAllowedAfterCommands(String functionName, String functionInvocation) throws Exception {
+
+        // Source commands
+        String function = functionName.toLowerCase(Locale.ROOT);
 
         // Processing commands
         assertEquals(
-            "1:43: [QSTR] function cannot be used after DISSECT",
-            error("from test | dissect first_name \"%{foo}\" | where qstr(\"Connection\")")
+            "1:43: [" + functionName + "] function cannot be used after DISSECT",
+            error("from test | dissect first_name \"%{foo}\" | where " + functionInvocation)
         );
-        assertEquals("1:27: [QSTR] function cannot be used after DROP", error("from test | drop emp_no | where qstr(\"Anna\")"));
         assertEquals(
-            "1:71: [QSTR] function cannot be used after ENRICH",
-            error("from test | enrich languages on languages with lang = language_name | where qstr(\"Anna\")")
+            "1:27: [" + functionName + "] function cannot be used after DROP",
+            error("from test | drop emp_no | where " + functionInvocation)
         );
-        assertEquals("1:26: [QSTR] function cannot be used after EVAL", error("from test | eval z = 2 | where qstr(\"Anna\")"));
         assertEquals(
-            "1:44: [QSTR] function cannot be used after GROK",
-            error("from test | grok last_name \"%{WORD:foo}\" | where qstr(\"Anna\")")
+            "1:71: [" + functionName + "] function cannot be used after ENRICH",
+            error("from test | enrich languages on languages with lang = language_name | where " + functionInvocation)
         );
-        assertEquals("1:27: [QSTR] function cannot be used after KEEP", error("from test | keep emp_no | where qstr(\"Anna\")"));
-        assertEquals("1:24: [QSTR] function cannot be used after LIMIT", error("from test | limit 10 | where qstr(\"Anna\")"));
         assertEquals(
-            "1:35: [QSTR] function cannot be used after MV_EXPAND",
-            error("from test | mv_expand last_name | where qstr(\"Anna\")")
+            "1:26: [" + functionName + "] function cannot be used after EVAL",
+            error("from test | eval z = 2 | where " + functionInvocation)
         );
         assertEquals(
-            "1:45: [QSTR] function cannot be used after RENAME",
-            error("from test | rename last_name as full_name | where qstr(\"Anna\")")
+            "1:44: [" + functionName + "] function cannot be used after GROK",
+            error("from test | grok last_name \"%{WORD:foo}\" | where " + functionInvocation)
         );
         assertEquals(
-            "1:52: [QSTR] function cannot be used after STATS",
-            error("from test | STATS c = COUNT(emp_no) BY languages | where qstr(\"Anna\")")
+            "1:31: [" + functionName + "] function cannot be used after KEEP",
+            error("from test | keep first_name | where " + functionInvocation)
```

Comment:
this should work for match `from test | keep first_name | where match(first_name, "Anna")`
the same if `match` is used in a `where` after a `drop`.